### PR TITLE
Added Crop Module

### LINKED ISSUE #5: image cropping module
A module which crops images according to passed width and height values.

### PR DESCRIPTION
**Pull request by ccpandhare**
I have implemented a simple crop module: `/src/modules/Crop.js`

(fixes #5)

It accepts params as:

```
options.X : The X-offset after which cropping starts | default 0
options.Y : The Y-offset after which cropping starts | default 0
options.W : Width of the crop | default 50% of image width
options.H : Height of the crop | default 50% of image height
options.format : format of image | default "png"
```

Also, Should we add dist/image-sequencer.js to .gitignore ?

### CODE REVIEW

File: src/modules/Crop.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+       pixels.shape = [w,h,4];
+       pixels.stride[1] = 4*w;
+
+       var buffer = base64.encode();
+       savePixels(pixels, options.format)
+         .on('end', function() {
+
+         var img = new Image();
+
+         img.src = 'data:image/' + options.format + ';base64,' + buffer.read().toString();
```
Comment (jywarren) [inline src/modules/Crop.js]: I think we should try to abstract this to make it easier to do from different modules, but will merge for now. Thanks!

**Comment by jywarren** [comment]
I'm not completely sure re: `dist/image-sequencer.js` -- i can see the benefit to asking others to compile it in, but the cleanliness of asking them not to. Do you think it's a good idea?

Also, I'd prefer lower-case parameters like `options.w` if that's OK with you -- would you mind rebasing and making that change?

I'm also thinking about how we might take a set of parameters required and provide (in the module) a description of what they should be and their default -- for use either by programmers, or by the HTML interface when you'd be able to configure the step as a user. Thoughts on this?

File: src/modules/Crop.js (hunk 1 above)
Comment (ccpandhare) [inline src/modules/Crop.js]: Yes you're right. I shall do that soon.

**Comment by ccpandhare** [comment]
Ohh @jywarren, I understood that now.

**Comment by ccpandhare** [comment]
@jywarren, are the parameters okay now?

File: src/modules/ImageSelect.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ module.exports = function ImageSelect(options) {
 
   // this module is unique because it creates the image
   function draw(image) {
-    options.el.html(image);
+    el.html(image);
```
Comment (jywarren) [inline src/modules/ImageSelect.js]: Do you mean to include these changes, or is this new commit based on something other than the latest `master` branch? Thanks!

File: src/modules/Crop.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Image Cropping module
```
Comment (jywarren) [inline src/modules/Crop.js]: Can you paste in the usage and defaults to the comment area here, until we figure out where to more appropriately store that info? Just the content from the body of your original PR. Thanks!

**Comment by jywarren** [comment]
Yes, I think this is good, pending a couple more tweaks I requested. I think it's going to relate to https://github.com/publiclab/image-sequencer/issues/11 -- which @rackstar17 is interested in -- for displaying a UI to set the crop options. Could be good to coordinate, or for @rackstar17 to attempt a UI for this module first?

**Comment by ccpandhare** [comment]
@jywarren  I think @rackstar17 should try this first, if he wants.
Else I have no problem doing it.

**Comment by jywarren** [comment]
This is really nice and clean. Shall I merge now, is it ready, do you think?

**Comment by ccpandhare** [comment]
Yes, I think this should be merged for now.
And as far as the Accepting input from UI issue is concerned, We'll have to update all modules after that.
For example, ImageThreshold.js could accept user input (the thresholding constant).

**Comment by jywarren** [comment]
Perfect, then. Agreed!

**Comment by ccpandhare** [comment]
@jywarren Since @rackstar17 isn't working on the UI, should I?

**Comment by jywarren** [comment]
Sure, that sounds great! Thank you.

**Pull request merged**